backend/src/Installer.ts

```diff
@@ -44,13 +45,11 @@ export default class Installer {
     this.dependencies = args.dependencies
   }
 
-  async check() {
+  async check(log?: boolean) {
     d('CHECK INSTALLATION', { name: this.name, version: this.version })
-    const current = await this.isCurrent()
-    current
-      ? EventBus.emit(Installer.EVENTS.installed, this.toJSON())
-      : EventBus.emit(Installer.EVENTS.notInstalled, this.name)
-    return current
+    if (await this.isCurrent(log)) return EventBus.emit(Installer.EVENTS.installed, this.toJSON())
+    if (environment.isElevated) return await binaryInstaller.install()
```

Comment:
Seems odd that a check function would do an install.  Recommend breaking into two functions (check and install) or renaming the function.

Reply:
So there is a hearbeat loop that the backend runs and just checks a few things: cli installation, connections and updates. 

Each model has a check function that it runs regularly. If the check needs an action the action is performed.

In terms of the installer check the, actions that it might perform are telling the user that they will be prompted for permission to install the system service, or to just do the installation in the background if they are running as root. 

The connections check will restart a connection that meets the criteria and the update check will notify the user of an available update.

Anyway, since it's a check on the installer I don't feel like it's wrong to have it trigger the installation, it's what it was designed to do. There is another function that is an install function that can be triggered manually. the binaryInstaller.install() method.

Hope that helps!

Reply:
That does thanks!